### ISSUE TITLE
BUG: SelectBox generates invalid markup

### ISSUE THREAD

**Issue by grebaldi** [neos/neos-ui#3826]
There is a least one occasion of the select box generating directly nested `<ul>`-tags, which is invalid markup.

I noticed it in the select box for node tree presets:

![Screenshot_2024-07-11_20-00-32 SelectBox vis](https://github.com/neos/neos-ui/assets/2522299/8dbd194e-4f65-46e3-b185-6b6085ce390c)

![Screenshot_2024-07-11_20-00-04 SelectBox markup](https://github.com/neos/neos-ui/assets/2522299/7f8069ae-a73c-42a2-a650-08c2fb63d54a)

There might be other occasions yet.

Affected Versions:
Neos UI: 8.3

**Comment by laurahaenel** [comment]
Also occurs in

- node tree presets
- dimension dropdown
- view/preview dropdown
- node type dropdown (inspector)
- [add more]

It currently works in

- user dropdown

**Comment by WuertzMariia** [comment]
Hi, 
I would like to fix this bug.
Thanks

**Comment by minedesign-jl** [comment]
@laurahaenel @WuertzMariia are any of you currently working on this? If not I might take a look at it.

**Comment by laurahaenel** [comment]
Hi, I dont think so, please take it! Thanks heaps

**Comment by minedesign-jl** [comment]
**[Update]**

Seems to only effect  SelectBox component. Ui Component SelectBox.js, uses drowdown.contents, which returns a `<ul>` and adds an additional `<ul>` wrapper.

Selects with double `<ul>` or direct children that are not `<li>`

- [x] node tree presets
- [x] workspace dropdown
- [x] dimension dropdown
- [x] view/preview dropdown - subheadlines are `div`s as direct children of `<ul>`
- [x] node type dropdown (inspector)
- [x] open graph type (inspector)
- [x] twitter card - card type (inspector)
- [x] xml sitemap (inspector)

Selects with **valid** markup
- user dropdown
- dropdown next to publish button
- media management 
-- filter & sort dropdowns
-- asset options dropdown
- site management - default asset collection (select)

additional proposed changes (issue #4140 created)
- turning Dropdown__btn should be `button` instead of div[role=button] --> to improved ax and keyboard usability 
-- switch children from `<div> `to `<span>` with display block for valid markup
- selectBox search box markup is invalid because`<input>` may not be descendent of an element with `role=button`
